src/control/server/server.go

```diff
@@ -265,13 +266,9 @@ func Start(log *logging.LeveledLogger, cfg *config.Server) error {
 		// If the configuration specifies that we should explicitly set
 		// hugepage values per engine instance, do it. Otherwise, let
 		// SPDK/DPDK figure it out.
-		if cfg.SetHugepages {
-			// If we have multiple engine instances with block devices, then
-			// apportion the hugepage memory among the instances.
-			engineCfg.Storage.Bdev.MemSize = hugePages.FreeMB() / len(cfg.Engines)
-			// reserve a little for daos_admin
-			engineCfg.Storage.Bdev.MemSize -= engineCfg.Storage.Bdev.MemSize / 16
-		}
+
+		engineCfg.NrHugepages = cfg.NrHugepages;
```

Comment:
it might be safer to read Free pages from hugePageInfo as by this point prepare (spdk setup.sh) will have been run and huge pages will have been allocated, so in the case that requested (NrHugePages) != allocated (getHugePageInfo().Free) you will have the accurate value.